homeassistant/components/tod/binary_sensor.py

```diff
@@ -144,15 +157,18 @@ def _naive_time_to_utc_datetime(self, naive_time):
         # calculate utc datetime corresponding to local time
         return dt_util.as_utc(datetime.combine(current_local_date, naive_time))
 
-    def _calculate_boundary_time(self):
+    def _calculate_boundary_time(self) -> None:
         """Calculate internal absolute time boundaries."""
+        if TYPE_CHECKING:
+            assert self._time_after is not None
+            assert self._time_before is not None
```

Comment:
This doesn't seem correct. It's `_calculate_boundary_time` sets these attributes after init.
```suggestion
```

Reply:
Right. That was a bit sloppy and now fixed